PyInstaller/building/makespec.py

```diff
@@ -263,6 +284,15 @@ def main(scripts, name=None, onefile=None,
         # The the text value 'None' means - use default icon.
         icon_file = 'None'
 
+    if datas:
+        # Convert the list with one string per `--add-data` option usage
+        # given by the argparser to a list with tuples
+        datas = [tuple(e.split(':')) for e in datas]
```

Comment:
Well, my idea was to do this in `add_data_or_binary` already. You can return a tuple there :-)